spec/02-integration/05-proxy/05-dns_spec.lua

```diff
@@ -206,5 +206,64 @@ for _, strategy in helpers.each_strategy() do
         assert.equals(0, assert(tonumber(stdout)))
       end)
     end)
+
+    describe("run in stream subsystem", function()
+      local domain_name = "www.example.test"
+      local address = "127.0.0.1"
+
+      local fixtures = {
+        dns_mock = helpers.dns_mock.new()
+      }
+      fixtures.dns_mock:A({
+        name = domain_name,
+        address = address,
+      })
+
+      lazy_setup(function()
+        local bp = helpers.get_db_utils(strategy, {
+          "routes",
+          "services",
+        })
+
+        local tcp_srv = bp.services:insert({
+          name = "tcp",
+          host = domain_name,
+          port = helpers.mock_upstream_stream_port,
+          protocol = "tcp",
+        })
+
+        bp.routes:insert {
+          destinations = {
+            { ip = "0.0.0.0/0", port = 19000 },
+          },
+          protocols = {
+            "tcp",
+          },
+          service = tcp_srv,
+        }
+
+        assert(helpers.start_kong({
+          database = strategy,
+          nginx_conf = "spec/fixtures/custom_nginx.template",
+          stream_listen = helpers.get_proxy_ip(false) .. ":19000",
+          log_level = "info",
+        }, nil, nil, fixtures))
+
+      end)
+
+      lazy_teardown(function()
+        helpers.stop_kong()
+      end)
+
+      it("resolve domain name", function()
+        local tcp = ngx.socket.tcp()
+        assert(tcp:connect(helpers.get_proxy_ip(false), 19000))
+        local MESSAGE = "echo, ping, pong. echo, ping, pong. echo, ping, pong.\n"
+        assert(tcp:send(MESSAGE))
+        local body = assert(tcp:receive("*a"))
+        assert.equal(MESSAGE, body)
+        tcp:close()
```

Comment:
I wonder what will happen if an assertion fails and the code does not reach this line. The [documentation](https://github.com/openresty/lua-nginx-module#tcpsockclose) suggests that:
> Socket objects that have not invoked this method (and associated connections) will be closed when the socket object is released by the Lua GC (Garbage Collector) or the current client HTTP request finishes processing.

So I think theoretically we should be fine. There's still potential for improper resource handling which might introduce flakiness but a lot of things have to go wrong for it to occur (i.e. the test would need to fail, GC would need to run late enough for the sockets to get exhausted... and then some other tests might fail).

Reply:
We must rely on the GC to work properly.  There is no need to complicate the test code to cover for the unlikely case that the assertion (and thus the test) fails.